packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -74,123 +90,522 @@ const propTypes = {
+  const {
+    ariaLabel,
+    buttonRefCallback,
+    inputAttributes,
+    inputRefCallback,
+    intl,
+    isIncomplete,
+    isInvalid,
+    name,
+    onBlur,
+    onButtonFocus,
+    onChange,
+    onClick,
+    onFocus,
+    onKeyDown,
+    placeholder,
+    required,
+    value,
+    ...customProps
+  } = props;
 
-    this.handleOnButtonClick = this.handleOnButtonClick.bind(this);
-    this.handleOnChange = this.handleOnChange.bind(this);
-    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+  const [date, dateDispatch] = useReducer(dateReducer, dateState);
+  const [isFocused, setFocused] = useState(false);
+  const [dayInitialFocused, setDayInitialFocused] = useState(false);
+  const [monthInitialFocused, setMonthInitialFocused] = useState(false);
+  const [yearInitialFocused, setYearInitialFocused] = useState(false);
+  const dayInputRef = useRef();
+  const monthInputRef = useRef();
+  const yearInputRef = useRef();
+
+  const id = customProps.id ? customProps.id : undefined;
+  if (customProps.id) {
+    delete customProps.id;
   }
 
-  componentDidUpdate(prevProps) {
-    if (this.shouldShowPicker && !prevProps.shouldShowPicker && this.props.onClick) {
-      this.props.onClick();
+  const { onCalendarButtonClick, shouldShowPicker } = customProps;
+  delete customProps.onCalendarButtonClick;
+  delete customProps.shouldShowPicker;
+
+  const additionalInputProps = { ...customProps, ...inputAttributes };
+  const momentDateFormat = DateUtil.getFormatByLocale(intl.locale);
+  const dateValue = DateUtil.convertToISO8601(value, momentDateFormat);
+  const buttonText = intl.formatMessage({ id: 'Terra.datePicker.openCalendar' });
+  const dateFormatOrder = intl.formatMessage({ id: 'Terra.datePicker.dateFormatOrder' });
+  const dateSeparator = intl.formatMessage({ id: 'Terra.datePicker.separator' });
+  const label = ariaLabel || intl.formatMessage({ id: 'Terra.datePicker.date' });
+  const placeholderValues = DateUtil.getDateInputValues(dateFormatOrder, placeholder, dateSeparator);
+  const theme = React.useContext(ThemeContext);
+
+  const setDayRef = useCallback(node => {
+    dayInputRef.current = node;
+    if (inputRefCallback && dateFormatOrder === 'YMD') {
+      inputRefCallback(node);
     }
-  }
+  }, [dateFormatOrder, inputRefCallback]);
 
-  handleOnButtonClick(event) {
-    const attributes = { ...this.props.inputAttributes };
+  const setMonthRef = useCallback(node => {
+    monthInputRef.current = node;
+  }, []);
 
-    if (!attributes.readOnly && this.onCalendarButtonClick && this.props.onClick) {
-      this.onCalendarButtonClick(event, this.props.onClick);
+  const setYearRef = useCallback(node => {
+    yearInputRef.current = node;
+    if (inputRefCallback && (dateFormatOrder === 'MDY' || dateFormatOrder === 'DMY')) {
+      inputRefCallback(node);
     }
-  }
+  }, [dateFormatOrder, inputRefCallback]);
+
+  useEffect(() => {
+    if (shouldShowPicker && onClick) {
+      onClick();
+    }
```

Comment:
Why does does this scenario trigger the onClick ? I see it's in the prev implementation. Could we add a comment on why?

Reply:
I tried to play around on dev site to trigger the onClick here but was unable to do so.
I also do not know the purpose of `shouldShowPicker`, maybe this can be removed?
cc: @benbcai

Reply:
Added documentation here : https://github.com/cerner/terra-framework/pull/1208/commits/3943529c5a8492e45c28ab32f4469a9e2329c14f